scipy/interpolate/_aaa.py

```diff
@@ -309,6 +319,72 @@ def _compute_weights(self, f, z, rtol, max_terms):
         self.support_values = fj[i_non_zero]
         self.weights = wj[i_non_zero]
         self.errors = errors[: m + 1]
+        self._points = z
+        self._values = f
+
+    def clean_up(self, cleanup_tol=1e-13):
+        """Automatic removal of Froissart doublets.
+
+        Parameters
+        ----------
+        cleanup_tol : float, optional
+            Poles with residues less than this number times the geometric mean
+            of `values` times the minimum distance to `points` are deemed spurious by
+            the cleanup procedure, defaults to 1e-13.
+        """
+        # Find negligible residues
+        if any(self._values):
+            geom_mean_abs_f = scipy.stats.gmean(
+                np.abs(self._values[np.nonzero(self._values)])
+            )
+        else:
+            geom_mean_abs_f = 0
+
+        Z_distances = np.min(
+            np.abs(np.subtract.outer(self.poles(), self._points)), axis=1
+        )
+
+        with np.errstate(divide="ignore", invalid="ignore"):
+            ii = np.flatnonzero(
+                np.abs(self.residues()) / Z_distances < cleanup_tol * geom_mean_abs_f
+            )
+
+        ni = ii.size
+        if ni == 0:
+            return
+
+        warnings.warn(f"{ni} Froissart doublets detected.", RuntimeWarning,
+                        stacklevel=2)
+
+        # For each spurious pole find and remove closest support point:
+        closest_spt_point = np.argmin(
+            np.abs(np.subtract.outer(self.support_points, self.poles()[ii])), axis=0
+        )
+        self.support_points = np.delete(self.support_points, closest_spt_point)
+        self.support_values = np.delete(self.support_values, closest_spt_point)
+
+        # Remove support points z from sample set
+        mask = np.logical_and.reduce(
+            np.not_equal.outer(self._points, self.support_points), axis=1
+        )
+        f = self._values[mask]
+        z = self._points[mask]
```

Comment:
Here a comment would be helpful to explain why you need to recompute the weights after removing the doublets. At least a link to the relevant section of the AAA paper.